pkg/audit/disqualification_test.go

```diff
@@ -103,3 +109,136 @@ func calcReputation(dossier *overlay.NodeDossier) float64 {
 
 	return alpha / (alpha + beta)
 }
+
+func TestDisqualifiedNodesGetNoDownload(t *testing.T) {
+
+	// - uploads random data
+	// - mark a node as disqualified
+	// - check we don't get it when we require order limit
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		upl := planet.Uplinks[0]
+
+		err := satellite.Audit.Service.Close()
+		require.NoError(t, err)
+
+		testData := make([]byte, 8*memory.KiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = upl.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		projects, err := satellite.DB.Console().Projects().GetAll(ctx)
+		require.NoError(t, err)
+		require.Len(t, projects, 1)
+
+		encScheme := upl.GetConfig(satellite).GetEncryptionScheme()
+		cipher := encScheme.Cipher
+		encryptedAfterBucket, err := streams.EncryptAfterBucket(ctx, "testbucket/test/path", cipher, &storj.Key{})
+		require.NoError(t, err)
+
+		lastSegPath := storj.JoinPaths(projects[0].ID.String(), "l", encryptedAfterBucket)
+		pointer, err := satellite.Metainfo.Service.Get(ctx, lastSegPath)
+		require.NoError(t, err)
+
+		disqualifiedNode := pointer.GetRemote().GetRemotePieces()[0].NodeId
+		disqualifyNode(t, ctx, satellite, disqualifiedNode)
+
+		request := overlay.FindStorageNodesRequest{
+			MinimumRequiredNodes: 4,
+			RequestedCount:       0,
+			FreeBandwidth:        0,
+			FreeDisk:             0,
+			ExcludedNodes:        nil,
+			MinimumVersion:       "", // semver or empty
+		}
+		nodes, err := satellite.Overlay.Service.FindStorageNodes(ctx, request)
+		require.True(t, overlay.ErrNotEnoughNodes.Has(err))
+
+		require.NotEmpty(t, nodes)
+		for _, node := range nodes {
+			reputable, err := satellite.Overlay.Service.IsVetted(ctx, node.Id)
```

Comment:
alternatively or in addition, you could check `require.NotEqual(t, node.Id, disqualifiedNode)`